pkgs/leak_tracker_flutter_testing/lib/src/leak_tracking_for_tests.dart

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:leak_tracker/leak_tracker.dart';
+import 'package:meta/meta.dart';
+
+void _emptyLeakHandler(Leaks leaks) {}
+
+/// Leak tracking settings and helper methods for tests.
+abstract class LeakTrackingForTests {
+  /// Current configuration for leak tracking.
+  static LeakTrackingForTestsSettings settings =
+      const LeakTrackingForTestsSettings();
+
+  /// Updates [settings] to pause leak tracking.
+  static void pause() => settings = paused();
+
+  /// Updates [settings] to start leak tracking.
+  static void start() => settings = started();
+
+  /// Creates a copy of current settings with [paused] set to true.
+  static LeakTrackingForTestsSettings paused() =>
+      settings.copyWith(paused: true);
```

Comment:
I did not use name `withPaused` and `withStarted` because this method will be used massively, to exempt individual tests from leak tracking.

After merging this PR, I will update flutter/test, to have 8% of tests that are not leak free yet looking like this: 

```
    testWidgets('initialTimerDuration falls within limit',
    experimentalLeakTracking: LeakTrackingForTests.paused(),
    (WidgetTester tester) async {
        ...
```

withPaused looks less readable.

Reply:
wdyt about replacing "paused" with "ignore" or "skip" and replacing "started" with "track"? Pause and start are confusing to me because this implies some sort of live and modifiable state, whereas it is actually a static setting defined by code. This would also be consistent with the "skip" and "track" terminology used elsewhere which is nice.

Reply:
+1 to this

Reply:
good idea